src/storage/mongodb/TagStorage.ts

```diff
@@ -228,4 +230,29 @@ export default class TagStorage {
       result: tagsMDB
     };
   }
+
+  public static async createOICPVirtualUserTags(tenantID: string): Promise<void> {
+    const defaultTagIds = [OICPDefaultTagId.RemoteIdentification, OICPDefaultTagId.QRCodeIdentification, OICPDefaultTagId.PlugAndChargeIdentification];
+    // eslint-disable-next-line @typescript-eslint/no-misused-promises
+    defaultTagIds.forEach(async (defaultTagId) => {
+      let tag = await this.getTag(tenantID, defaultTagId);
+      if (!tag) {
+        const virtualOICPUser = await UserStorage.getOICPVirtualUser(tenantID);
+        tag = {
+          id: defaultTagId,
+          userID: virtualOICPUser.id,
+          user: virtualOICPUser,
+          description: 'Badge for OICP Remote Authorization',
+          issuer: false,
+          active: true,
+          createdOn: new Date(),
+          default: false
+        } as Tag;
+      } else {
+        tag.active = true;
+      }
+      // Save
+      await TagStorage.saveTag(tenantID, tag);
```

Comment:
I do not understand why you need to create and store tag for the OICP virtual user. 
If the authorisation code is able to cope with roaming platform remote authorisation properly when needed (and only when needed to avoid to slow down to most common authorisation scheme), you then do not need to add that kind of hack.

Reply:
Ok, I understand it. I changed the code so that there is no need for extra Tags for the virtual user.